Analysis/Tasks/PWGCF/GenericFramework.cxx

```diff
@@ -0,0 +1,195 @@
+
+  O2_DEFINE_CONFIGURABLE(cfgEfficiency, std::string, "", "CCDB path to efficiency object")
+  O2_DEFINE_CONFIGURABLE(cfgAcceptance, std::string, "", "CCDB path to acceptance object")
+
+  ConfigurableAxis axisVertex{"axisVertex", {20, -10, 10}, "vertex axis for histograms"};
+  ConfigurableAxis axisPhi{"axisPhi", {60, 0.0, 2.*M_PI}, "phi axis for histograms"};
+  ConfigurableAxis axisEta{"axisEta", {40, -1., 1.}, "eta axis for histograms"};
+  ConfigurableAxis axisPt{"axisPt", {VARIABLE_WIDTH, 0.2, 0.25, 0.30, 0.40, 0.45, 0.50, 0.55, 0.60, 0.65, 0.70, 0.75, 
+                        0.80, 0.85, 0.90, 0.95, 1.00, 1.10, 1.20, 1.30, 1.40, 1.50, 1.60, 1.70, 1.80, 1.90,
+                        2.00, 2.20, 2.40, 2.60, 2.80, 3.00}, "pt axis for histograms"};
+  ConfigurableAxis axisMultiplicity{"axisMultiplicity", {VARIABLE_WIDTH, 0, 5, 10, 20, 30, 40, 50, 60, 70, 
+                        80, 90, 100.1}, "multiplicity / centrality axis for histograms"};
+
+  Filter collisionFilter = nabs(aod::collision::posZ) < cfgCutVertex;
+  Filter trackFilter = (nabs(aod::track::eta) < cfgCutEta) && (aod::track::pt > cfgCutPtMin) 
+  && (aod::track::pt < cfgCutPtMax) && ((aod::track::isGlobalTrack == (uint8_t) true) 
+  || (aod::track::isGlobalTrackSDD == (uint8_t) true));
+  using myTracks = soa::Filtered<soa::Join<aod::Tracks, aod::TrackSelection>>;
+
+  //Connect to ccdb
+  Service<ccdb::BasicCCDBManager> ccdb;
+  Configurable<long> nolaterthan{"ccdb-no-later-than", std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(), "latest acceptable timestamp of creation for the object"};
+  Configurable<std::string> url{"ccdb-url", "http://ccdb-test.cern.ch:8080", "url of the ccdb repository"};
+
+  struct Config {
+    TH1D* mEfficiency = nullptr;
+    GFWWeights* mAcceptance = nullptr;
+  } cfg;
+
+  //Define output
+  OutputObj<FlowContainer> fFC{FlowContainer("FlowContainer")};
+  HistogramRegistry registry{"registry"};
+
+
+  // define global variables
+  GFW *fGFW;
+  std::vector<GFW::CorrConfig> corrconfigs;
+  TRandom* fRndm;
+
+  void init(InitContext const&)
+  {
+
+    ccdb->setURL(url.value);
+    ccdb->setCaching(true);
+    ccdb->setCreatedNotAfter(nolaterthan.value);
+
+    //Global effiencies
+    if (cfgEfficiency.value.empty() == false) {
+      cfg.mEfficiency = ccdb->getForTimeStamp<TH1D>(cfgEfficiency.value, nolaterthan.value);
+      if(cfg.mEfficiency) LOGF(info, "Loaded efficiency histogram %s (%p)", cfgEfficiency.value.c_str(), (void*)cfg.mEfficiency);
+      else LOGF(info, "Loaded efficiency histogram %s (%p)", cfgEfficiency.value.c_str(), (void*)cfg.mEfficiency);
+    }
+
+
+    registry.add("hPhi","",{HistType::kTH1D,{axisPhi}});
+    registry.add("hEta","",{HistType::kTH1D,{axisEta}});
+    registry.add("hVtxZ","",{HistType::kTH1D,{axisVertex}});
+    Int_t pows[7] = {3,0,2,2,3,3,3}; 
+    Int_t powsFull[7] = {5,0,4,4,3,3,3};
+
+    fRndm = new TRandom(0);
```

Comment:
I think TRandom should not be used (is this used like this in AliPhysics?). The ROOT documentation writes:
`Please note also that this class (TRandom) implements also a very simple generator (linear congruential) with period = 109, known to have defects (the lower random bits are correlated) and it is failing the majority of the random number generator tests. Therefore it should NOT be used in any statistical study.`
https://root.cern.ch/doc/master/classTRandom.html